dotnet/targets/Xamarin.Shared.Sdk.targets

```
@@ -1115,7 +1115,7 @@
 				In such setup, during runtime, AOT images are loaded but marked as unusuable as they are compiled with `full` compiler switch and the code falls back to interpreter.
 				Dedup AOT image is specially handled by the AOT runtime and it is not allowed to have it marked as unusuable.
 			-->
-			<_IsDedupEnabled Condition="'$(_IsDedupEnabled)' == '' And '$(_RunAotCompiler)' == 'true' And '$(IsMacEnabled)' == 'true' And '$(TargetArchitectures)' == 'ARM64'">true</_IsDedupEnabled>
+			<_IsDedupEnabled Condition="'$(_IsDedupEnabled)' == '' And '$(_RunAotCompiler)' == 'true' And '$(IsMacEnabled)' == 'true' And '$(TargetArchitectures)' == 'ARM64' And '$(MtouchInterpreter)' != 'all'">true</_IsDedupEnabled>
```

```suggestion
			<_IsDedupEnabled Condition="'$(_IsDedupEnabled)' == '' And '$(_RunAotCompiler)' == 'true' And '$(IsMacEnabled)' == 'true' And '$(TargetArchitectures)' == 'ARM64' And '$(MtouchInterpreter)' != 'all' And '$(Configuration)' == 'Release'">true</_IsDedupEnabled>
```
Should we enable it only for release builds?

---

@kotlarmilos are there ever any cases/configurations where the dedup optimization is required for correctness?

---

No, this is just an optimization.

---

I think I'd like to avoid changing the build settings depending on the configuration if we don't really have to, because it feels rather random for users if they use other configurations than "Debug" and "Release" - see https://github.com/dotnet/sdk/issues/31918.

It'll end up being one knob to turn to diagnose "my app crashes when I release it, but works fine locally"